mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutor.java

```diff
@@ -174,6 +175,11 @@ public TaskExecutor(
             Executors.newFixedThreadPool(
                 Hardware.getNumberCPUCores(),
                 new ExecutorThreadFactory("taskexecutor-io"));
+
+        this.runtimeTaskExecutor =
+            Executors.newFixedThreadPool(
+                1,
```

Comment:
If you merge this, I recommend increasing the number of threads to more than one.